src/commands/lambda/instrument.ts

```diff
@@ -286,20 +286,54 @@ export class InstrumentCommand extends Command {
     }
 
     if (willUpdate) {
-      const promises = Object.values(configGroups).map((group) =>
-        updateLambdaFunctionConfigs(group.lambda, group.cloudWatchLogs, group.configs)
-      )
-      const spinner = renderer.updatingFunctionsSpinner(promises.length)
-      spinner.start()
-      try {
-        await Promise.all(promises)
-        spinner.succeed(renderer.renderUpdatedLambdaFunctions(promises.length))
-      } catch (err) {
-        this.context.stdout.write(renderer.renderFailureDuringUpdateError(err))
-        spinner.fail(renderer.renderFailedUpdatingLambdaFunctions())
+      const totalFunctions = Object.values(configGroups).reduce((c, group) => (c += group.configs.length), 0)
+      let totalFailedUpdates = 0
+      for (const group of configGroups) {
+        const spinner = renderer.updatingFunctionsConfigFromRegionSpinner(group.region, group.configs.length)
+        spinner.start()
+        const failedUpdates = []
+        for (const config of group.configs) {
+          try {
+            await updateLambdaFunctionConfig(group.lambda, group.cloudWatchLogs, config)
+          } catch (err) {
+            failedUpdates.push({functionARN: config.functionARN, error: err})
+            totalFailedUpdates += 1
+          }
+        }
+
+        if (failedUpdates.length === group.configs.length) {
```

Comment:
Seems we're processing failed updates both in the inner `group.configs` loop, but also in the outer `configGroups` loop, where we `renderFail` on lines 322 and 323. Is there an opportunity to combine and simplify the logic here?

I'm a bit unclear on the difference in this block than the lower block.

Reply:
One is to state errors for the current region, and the other one for all the instrumented functions overall. 

For example here. Where we expect 3 functions to be updated but one fails.
<img width="881" alt="Screenshot 2023-02-03 at 2 57 23 PM" src="https://user-images.githubusercontent.com/30836115/216698337-a976ebfd-c8cd-403b-a0f0-1d816b890ed3.png">

It might be a little verbose if you are instrumenting for only one region.